modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2834,6 +2834,23 @@ def _groupby_internal_columns(self, by, drop):
     groupby_skew = GroupbyReduceImpl.build_qc_method("skew")
 
     def groupby_mean(self, by, axis, groupby_kwargs, agg_args, agg_kwargs, drop=False):
+        if ExperimentalGroupbyImpl.get():
```

Comment:
What is the principle by which methods are chosen in which we try to use an experimental implementation?

Reply:
All of the groupby methods should have the ability to be executed in an experimental mode (I guess I've added support for all of them, correct me if I'm wrong). For now, we give the right for a user to decide where they want to apply this new implementation as I haven't built a strong heuristic yet that would optimally switch between all of our groupby implementations.

From my experience, new implementation works better for:
1. Full-axis groupbys when the number of rows 1mln+
2. MapReduce groupbys when there's a huge amount of groups
3. MapReduce groupbys when there are more than 10mln rows

Reply:
Needs to be reflected in docs btw

Reply:
> All of the groupby methods should have the ability to be executed in an experimental mode (I guess I've added support for all of them, correct me if I'm wrong). For now, we give the right for a user to decide where they want to apply this new implementation as I haven't built a strong heuristic yet that would optimally switch between all of our groupby implementations.
> 
> From my experience, new implementation works better for:
> 
> 1. Full-axis groupbys when the number of rows 1mln+
> 2. MapReduce groupbys when there's a huge amount of groups
> 3. MapReduce groupbys when there are more than 10mln rows

I got it. @dchigarev thanks for the clarification.

Reply:
I'll do it in a separate PR (created a tracker for this: https://github.com/modin-project/modin/issues/6021)